File: src/main/kotlin/com/full/stack/assignment/f1/data/cache/mapper/SeasonMapper.kt
package com.full.stack.assignment.f1.data.cache.mapper

import com.full.stack.assignment.f1.data.cache.entity.SeasonEntity
import com.full.stack.assignment.f1.model.Season
import org.springframework.stereotype.Component

@Component
class SeasonMapper(
    private val driverMapper: DriverMapper,
) : Mapper<Season, SeasonEntity> {
    override fun toDomain(entity: SeasonEntity): Season {
        return Season(
            year = entity.year,
            champion = driverMapper.toDomain(entity.champion),
        )
    }

    override fun toEntity(domain: Season): SeasonEntity {
        return SeasonEntity(
            year = domain.year,
            champion = driverMapper.toEntity(domain.champion),
        )
    }
}

File: frontend/lib/data/remote/mappers/season_mapper.dart
import 'package:f1_api_client/api.dart';
import '../../../domain/models/season/season.dart';
import '../../../helpers/mapper/mapper.dart';
import 'driver_mapper.dart';

class SeasonMapper extends Mapper<SeasonDTO, Season, void> {
  final DriverMapper _driverMapper;

  SeasonMapper(this._driverMapper);

  @override
  Season map(SeasonDTO param, {void metadata}) {
    final champion = param.champion;
    return Season(year: param.year, champion: _driverMapper.map(champion));
  }
}
